apps/src/componentLibrary/dropdown/_CustomDropdown.tsx

```diff
@@ -0,0 +1,147 @@
+import React, {
+  useCallback,
+  useMemo,
+  useRef,
+  useEffect,
+  KeyboardEvent,
+} from 'react';
+import classNames from 'classnames';
+
+import {
+  ComponentSizeXSToL,
+  DropdownColor,
+} from '@cdo/apps/componentLibrary/common/types';
+import moduleStyles from './customDropdown.module.scss';
+
+import FontAwesomeV6Icon, {
+  FontAwesomeV6IconProps,
+} from '@cdo/apps/componentLibrary/fontAwesomeV6Icon';
+import {useDropdownContext} from '@cdo/apps/componentLibrary/common/contexts/DropdownContext';
+import {dropdownColors} from '@cdo/apps/componentLibrary/common/constants';
+
+export interface CustomDropdownProps {
+  /** CheckboxDropdown name */
+  name: string;
+  /** CheckboxDropdown color */
+  color?: DropdownColor;
+  /** CheckboxDropdown size */
+  size: ComponentSizeXSToL;
+  /** CheckboxDropdown disabled state */
+  disabled?: boolean;
+  /** CheckboxDropdown label */
+  labelText: string;
+  /** Does custom dropdown hase a selected value (Renders a checkmark icon in the dropdown button if true) */
+  isSomeValueSelected?: boolean;
+  /** Custom icon to show for the dropdown button*/
+  icon?: FontAwesomeV6IconProps;
+  /** Children */
+  children: React.ReactNode;
+}
+
+/**
+ * Design System: _CustomDropdown Component.
+ * This is a PRIVATE component that is used to create different custom dropdowns.
+ * Only used inside of Design System components. DO NOT IMPORT OR USE DIRECTLY.
+ */
+const CustomDropdown: React.FunctionComponent<CustomDropdownProps> = ({
```

Comment:
Slick refactor! Wondering if a name like `BaseDropdown` might better communicate the role of this component?

Reply:
Good question.
I decided to call it a `_CustomDropdown` so that it'll be obvious that it's opposite of our `SImpleDropdown`  (which uses native hml select element for dropdown) which I also plan to move to dropdown folder so that all the dropdowns will be in one place. Giving all of this I don't have a strong opinion on this one, but leaning towards keeping it a `_CustomDropdown` since I believe `BaseDropdown` might cause some confusions in the future

Reply:
FWIW I also feel like something like `BaseDropdown` or `DropdownInternal` might be clearer, but not a strong opinion if you'd like to keep it `_CustomDropdown`

Reply:
Gotcha, will think about it and make some update in a follow up PR once `SimpleDropdown` will also be moved to `dropdown` folder